applications/StructuralMechanicsApplication/custom_constitutive/plane_stress_d_plus_d_minus_damage_masonry_2d.cpp

```diff
@@ -0,0 +1,688 @@
+	const double c3 = r_material_properties["BEZIER_CONTROLLER_C3"];
+	const double Gc = r_material_properties["FRACTURE_ENERGY_COMPRESSION"];
+	
+	// Calculate missing Bezier Determinators
+	const double alpha = 2.0 * (ep - (sp/E));
+	const double e0 = s0 / E;
+	const double ei = sp / E;
+	const double sk = sr + (sp - sr) * c1;
+	const double ej = ep + alpha * c2;
+	const double ek = ej + alpha * (1.0 - c2);
+	const double er = ( (ek - ej) * (sp - sr)/(sp -sk) ) + ej;
+	const double eu = er * c3;
+	const double gc = Gc / CharacteristicLength;
+	
+	// Perform the Energy Regularization of the Bezier Determinators
+	double gc_bezier; 
+	DamageDPlusDMinusMasonry2DLaw::RegulateBezierDeterminators(gc, sp, sk, sr, ep, ej, ek, er, eu);
+	
+	// Compute rDamage
+	double StrainLikeCounterpart = UniaxialStress / E;
+	double DamageVariableBezier = UniaxialStress;
+	if (StrainLikeCounterpart <= ep){
+		DamageVariableBezier = EvaluateBezierCurve(StrainLikeCounterpart, e0, ei, ep, s0, sp, sp);
+	} else {if (StrainLikeCounterpart <= ek){
+			DamageVariableBezier = EvaluateBezierCurve(StrainLikeCounterpart, ep, ej, ek, sp, sp, sk);
+			} else {if (StrainLikeCounterpart <= eu){
+				DamageVariableBezier = EvaluateBezierCurve(StrainLikeCounterpart, ek, er, eu, sk, sr, sr);
+				} else {
+					DamageVariableBezier = sr;
+					   }
+				   }
+		   }
+	rDamage = 1.0 - DamageVariableBezier / UniaxialStress;
+}
+/***********************************************************************************/
+/***********************************************************************************/
+void DamageDPlusDMinusMasonry2DLaw::RegulateBezierDeterminators(
+	const double specific_dissipated_fracture_energy,
+	const double sp, const double sk, const double sr, const double ep,
+	double& ej, double& ek, double& er, double& eu)
+{	
+	const double bezier_energy_1 = sp * ep / 2.0;
+	double bezier_energy_2
+	double bezier_energy_3
+	DamageDPlusDMinusMasonry2DLaw::ComputeBezierEnergy(bezier_energy_2, ep, ej, ek, sp, sp, sk);
+	DamageDPlusDMinusMasonry2DLaw::ComputeBezierEnergy(bezier_energy_3, ek, er, eu, sk, sr, sr);
+	const double BezierEnergy = bezier_energy_1 + bezier_energy_2 + bezier_energy_3;
+	
+	const double BezierStretcher = ((specific_dissipated_fracture_energy - bezier_energy_1) / 
+								   (BezierEnergy - bezier_energy_1)) - 1.0;
+	KRATOS_ERROR_IF(BezierStretcher <= -1.0) << "FRACTURE_ENERGY_COMPRESSION is too low, increase it to avoid constitutive snap-back!" << std::endl;	
+								   
+    // Update Strain values
+	ej = ej + BezierStretcher * (ej - ep);
+	ek = ek + BezierStretcher * (ek - ep);
+	er = er + BezierStretcher * (er - ep);
+	eu = eu + BezierStretcher * (eu - ep);
+}
+/***********************************************************************************/
+/***********************************************************************************/
+void DamageDPlusDMinusMasonry2DLaw::ComputeBezierEnergy(
+	double& BezierG,

```

Comment:
since this is reference (you use `&`) the name of the variable should be **r**BezierG